### ISSUE TITLE
unable to use unocss-icons from CDN in nuxt production.

### ISSUE THREAD

**Issue by Sandros94** [jd-solanki/anu#155]
I was testing stuff for #154 (I'm using that stackblitz reproduction) and I've noticed that when I use unocss icons from CDNs they don't get applied to Anu's components while in production.

dev (`npm run dev`):
![image](https://user-images.githubusercontent.com/13056429/230747171-69398e87-14f4-4d24-924d-aa599e01db3c.png)

production (`npm run build && npm run preview`):
![image](https://user-images.githubusercontent.com/13056429/230747127-c2835ed0-66f6-469c-9989-9c0e9de74992.png)

TBH I've still never tried Anu in production, so Idk if this was existing before v0.13.0

**Comment by jd-solanki** [comment]
Hi, @Sandros94 I tried [Anu demo](https://github.com/jd-solanki/anu-demo) repo in production and it's working as expected. Can you please share your repo?

**Comment by Sandros94** [comment]
Just woke up, but in the demo repo I can see that none of the icons comes from CDN.

I'll update my test repo in a bit.

**Comment by Sandros94** [comment]
Ok I've prepared a reproduction in my [test repo](https://github.com/Sandros94/test-nuxt3app/tree/Anu) under Anu branch.

Pull it, then `npm i && npm run build && npm run preview`

The key factor is that I'm pulling the icons from a cdn instead of installing them as packages:

https://github.com/Sandros94/test-nuxt3app/blob/ef447e42a721391562875efefc043368b818439b/unocss.config.ts#L23-L29

**Comment by jd-solanki** [comment]
Hey @Sandros94 

Can you please add lock file to your test repo? I'm getting errors due to missing lock file.

Awaiting your response.

**Comment by Sandros94** [comment]
Once I get back home I'll push that specific lockfile.

> Moreover, if possible can you please check if you are facing the same with unocss only (without anu)?

I didn't test unoccs only in that repo, but two notes about it:
- in other repos, where I do have unocss only with the same test config (minus anu oc), I've never faced something similar.
- In the second screenshot you can see the icon on the left that is just inside a `div`, and works correctly after `build`, so I cannot imagine how could it be unocss

**Comment by Sandros94** [comment]
While fixing a small error (the clock icon was white on white for the light theme), it made me discover a thing.

To solve #135 I decided to implement [this](https://github.com/jd-solanki/anu/issues/135#issuecomment-1483891668):
```vue
<template>
    <ClientOnly>
        <template #fallback>
            <ASwitch v-model="off" off-icon="i-svg-spinners:clock"/>
        </template>
        <ASwitch v-model="switchTheme" off-icon="i-ph:sun" on-icon="i-ph:moon"/>
    </ClientOnly>
</template>

<script setup>
const colorMode = useColorMode();
const off = ref(false)
const switchTheme = computed({
    get: () => colorMode.value === 'dark',
    set: (value) => {
        if (colorMode.preference === 'system') {
            // Set the opposite of the current system value
            colorMode.preference = value ? 'dark' : 'light';
        } else {
            // Set the preference back to 'system'
            colorMode.preference = 'system';
        }
    }
});
</script>
```

and as described in the closing comment of that issue I had to use the built-in `<ClientOnly>` nuxt component to make `ASwitch` load its state correctly since I couldn't know until the browser fully loaded the page. But I wanted to add the `fallback` so that the switch didn't just appear from nowhere.

# Here is my discovery
on `npm run build`, while the component is still on server side, it correctly loads as:
```vue
<template>
    <ASwitch v-model="off" off-icon="i-svg-spinners:clock"/>
</template>
```
but once it gets loaded in a browser (and thus get the information of the current theme), it should become:
![image](https://user-images.githubusercontent.com/13056429/233373117-6565ee4e-8f06-4d56-bf97-bd1018caedb8.png)
![image](https://user-images.githubusercontent.com/13056429/233373746-384be046-4618-479b-ae34-b10bc25508db.png)
```vue
<template>
    <ASwitch v-model="switchTheme" off-icon="i-ph:sun" on-icon="i-ph:moon"/>
</template>
```
but instead it becomes:
![image](https://user-images.githubusercontent.com/13056429/233374195-30b49b4d-c0c7-48d5-944e-333a4bba48c3.png)
![image](https://user-images.githubusercontent.com/13056429/233374383-e2781683-d9f7-4942-bd23-965a49796178.png)
```vue
<template>
    <ASwitch v-model="switchTheme" off-icon="i-svg-spinners:clock"/>
</template>
```

Basically when it switches from the `fallback` to the working one, it does not update the props too, but works correctly because it is using the right `v-model`.

# Current considerations
Later I'll update the test page in demo-branch with all three switch combination and create a new reply in this issue, to better describe what they are and potentially close this issue (or at least update the title). Because now I have come to the conclusion that yes, Anu is indeed using the unocss-icons from CDN, but it gets broken along the way once you use them in the built-in `<ClientOnly>` nuxt component when used with a `fallback`.

**Comment by jd-solanki** [comment]
Due to some issues, action wasn't able to make a edge release to test the latest release I said about. I will let you know once edge release will continue to work.

**Comment by Sandros94** [comment]
I started a [discussion](https://github.com/nuxt/nuxt/discussions/20419) in the Nuxt repo.
Maybe it's not an Anu issue at all.

**Comment by jd-solanki** [comment]
Hey @Sandros94 

What's the status of this issue? Did you manage to resolve the issue?

**Comment by Sandros94** [comment]
Unfortunately I haven't, and by the looks of it it seems that the nuxtlab team is busy with other developments.

One thing I wanted to try was to add a `key` prop to both, so that I could force vue to distinguish the main element from the fallback one.
